src/trainer/JasmineGraphTrainingSchedular.cpp

```diff
@@ -0,0 +1,137 @@
+    for (vector<vector<pair<string, string>>>::iterator i = result.begin(); i != result.end(); ++i) {
+        int count = 0;
+        string hostID;
+        string hostname;
+        for (std::vector<pair<string, string>>::iterator j = (i->begin()); j != i->end(); ++j) {
+            if (count == 0) {
+                hostID = j->second;
+            } else {
+                hostname = j->second;
+                hostData.push_back(pair<string, string>(hostID, hostname));
+            }
+            count++;
+        }
+    }
+    string id_partition = "";
+    string vertexCount = "";
+    string centralvertexCount = "";
+    int partition_id;
+    int vertexcount;
+    int centralVertexCount;
+
+    for (std::vector<pair<string, string>>::iterator j = (hostData.begin()); j != hostData.end(); ++j) {
+        sqlStatement =
+                "SELECT idpartition,vertexcount,central_vertexcount,graph_idgraph FROM partition INNER JOIN host_has_partition "
+                "ON partition.idpartition = host_has_partition.partition_idpartition WHERE graph_idgraph = " + graphID +
+                " AND host_idhost = "
+                + j->first;
+        std::vector<vector<pair<string, string>>> results = refToSqlite.runSelect(sqlStatement);
+
+        std::map<int, int> vertexCountToPartitionId;
+        cout << sqlStatement << endl;
+        for (std::vector<vector<pair<string, string>>>::iterator i = results.begin(); i != results.end(); ++i) {
+            std::vector<pair<string, string>> rowData = *i;
+
+            id_partition = rowData.at(0).second;
+            vertexCount = rowData.at(1).second;
+            centralvertexCount = rowData.at(2).second;
+
+            partition_id = stoi(id_partition);
+            vertexcount = stoi(vertexCount);
+            centralVertexCount = stoi(centralvertexCount);
+
+            //ToDO:get triangle count by partition id
+
+            vertexCountToPartitionId.insert({vertexcount + centralVertexCount, partition_id});
+
+
+        }
+
+        float totMemEstimation = 0;
+        int iteration = 0;
+        for (auto i = vertexCountToPartitionId.begin(); i != vertexCountToPartitionId.end(); ++i) {
+            totMemEstimation += estimateMemory(i->first);
+            if (totMemEstimation < 600000000) {  //total=65843492
+                iteration++;
+                partitionSequence[j->second].push_back(i->second);
+            } else {
+                partitionSequenceIteration[j->second].push_back(iteration);
+                iteration = 0;
+                totMemEstimation = 0;
+                partitionSequence[j->second].push_back(i->second);
+                iteration++;
+                totMemEstimation += estimateMemory(i->first);
+            }
+        }
+    }
+
+
+}
+
+float JasmineGraphTrainingSchedular::estimateMemory(int vertexCount) {
+    float featureMetrixSize;
+    float adjacencyMetrixSize;
+    float degreeMetrixSize;
+    float embeddingMetrixSize;
+    int featureCount = 100;
```

Comment:
Are we using 100 features every round of training? Can this be taken as a function argument?

Reply:
feature count will be taken from the database. Dinika will add it once this is merged

Reply:
Ok